### PR TITLE
Feature/unit testing

### PR DESCRIPTION
**Pull request by victoroicr**
* Commons module
* Integration Testing module
* User serialization/deserialization unit test

### CODE REVIEW

**Comment by briandoconnor** [comment]
@victoroicr @denis-yuen is this ready to go?  Do you need another person to test?

File: dockstore-client/src/test/java/io/dockstore/client/ClientTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.dockstore.client;
+
+/**
+ *
+ * @author xliu
+ */
+public class ClientTest {
```
Comment (denis-yuen) [inline dockstore-client/src/test/java/io/dockstore/client/ClientTest.java]: This class seems somewhat pointless, delete it

File: dockstore-client/src/test/java/io/dockstore/client/WebClientTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.dockstore.client;
+
+/**
+ *
+ * @author xliu
+ */
+public class WebClientTest {
```
Comment (denis-yuen) [inline dockstore-client/src/test/java/io/dockstore/client/WebClientTest.java]: same here

File: dockstore-common/src/main/java/io/dockstore/common/Constants.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    public static final String POSTGRES_PASSWORD = "database.postgresPass";
+    public static final String POSTGRES_DBNAME = "database.postgresDBName";
+    public static final String POSTGRES_MAX_CONNECTIONS = "database.maxConnections";
+
+    public static final String RABBIT_HOST = "rabbit.rabbitMQHost";
+    public static final String RABBIT_USERNAME = "rabbit.rabbitMQUser";
+    public static final String RABBIT_PASSWORD = "rabbit.rabbitMQPass";
+    public static final String RABBIT_QUEUE_NAME = "rabbit.rabbitMQQueueName";
+
+    public static final String PROVISION_MAX_RUNNING_CONTAINERS = "provision.max_running_containers";
```
Comment (denis-yuen) [inline dockstore-common/src/main/java/io/dockstore/common/Constants.java]: These variables seem copied from consonance, please delete all the ones that are not relevant to dockstore

**Comment by denis-yuen** [comment]
dockstore-integration-testing/src/test/resources/.run-fox.yml.swo and
dockstore-integration-testing/src/test/resources/.run-fox.yml.swp are checked in, but are probably vim temporary files. Please delete them

File: dockstore-integration-testing/src/test/resources/dockstore.yml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+template: Hello, %s!
+defaultName: Stranger
+quayClientID: 4CFMQRAV6121BVCG3SXA
+quayRedirectURI: http://10.0.29.49:8080/static/quay_callback.htm
+githubClientID: d2a5a83a0f0871d29540
+githubClientSecret: 0db285c4cb33c6edd30c43d51c4947b3f1e37f82
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/resources/dockstore.yml]: are these valid credentials?

File: dockstore-common/src/main/java/io/dockstore/common/Constants.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ * 
+ * This describes all keys available in Dockstore config files.
+ *
+ * @author xliu
+ */
+public class Constants {
+    public static final String WEBSERVICE_BASE_PATH = "webservice.base_path";
+    public static final String WEBSERVICE_TOKEN = "webservice.token";
+    public static final String WEBSERVICE_TOKEN2 = "webservice.token2";
+    public static final String WEBSERVICE_EXTRA_FILES = "webservice.extra_files";
```
Comment (denis-yuen) [inline dockstore-common/src/main/java/io/dockstore/common/Constants.java]: Is this one actually used for dockstore (or consonance)? 
Also in your next branch, change webservice.token2 into something more descriptive or add a comment to this file.

**Pull request merged**